src/ImageSegmentation.jl

```diff
@@ -2,18 +2,32 @@ __precompile__()
 
 module ImageSegmentation
 
-using Images, DataStructures, StaticArrays
+using Images, DataStructures, StaticArrays, ImageFiltering
+
+# For efficient hashing of CartesianIndex
+if !isdefined(Base.IteratorsMD, :cartindexhash_seed)
```

Comment:
is this needed if your minimum julia version is 0.6?

Reply:
Should I remove it from here and send a PR to `Compat`?

Reply:
@timholy thoughts? Would this break anything in any way to start getting different hash values for these? If it's otherwise a performance-only improvement, would it make sense for this to be available in a centralized, approximately-always-loaded place instead of only being changed when ImageSegmentation is loaded?

Reply:
It's just a performance tweak (though one with substantial consequences). I'd support moving it to Compat, though I don't think type piracy that leads only to performance changes is all that much to worry about. (Yes, better in Compat, but no crime having it only here.)